### PR TITLE
support SPARC detectors with roles numbered

### PR DESCRIPTION
**Pull request by pieleric**

### CODE REVIEW

File: src/odemis/gui/conf/data.py [hunk 1]
```diff
@@ -512,7 +450,7 @@
                 "accuracy": 3,
             },
         },
-    "photo-detector0":  # TODO: for every photo-detector* => make it a regex
+    r"photo-detector.*":
```
Comment (Sabrina85) [inline src/odemis/gui/conf/data.py]: no need for the `$` here?

File: src/odemis/gui/cont/streams.py [hunk 2]
```diff
@@ -2383,19 +2385,18 @@ def _createAddStreamActions(self):
...
             self.add_action("Angle-resolved", self.addAR)
 
         # On the SPARCv2, there is potentially 4 different ways to acquire a
-        # spectrum: two spectrographs, with each two ports. In practice, there
-        # are never more than 2 at the same time.
-        sptms = [main_data.spectrometer, main_data.spectrometer_int]
-        sptms = [s for s in sptms if s is not None]
-        for sptm in sptms:
-            if len(sptms) == 1:
+        # spectrum: two spectrographs, with each two ports.
```
Comment (Sabrina85) [inline src/odemis/gui/cont/streams.py]: keep it generic regarding number of spectrographs? Maybe the next system has 3 :P
word order: `each with two ports`

File: src/odemis/gui/conf/data.py [hunk 3]
```diff
@@ -907,6 +811,29 @@ def get_stream_settings_config():
...
 
+def get_hw_config(hw_comp, hw_settings):
+    """
+    Find the VA config for the given component.
+
+    hw_comp (HwComponent): The component to look at
+    hw_settings (dict): the hardware settings, as received from get_hw_settings_config()
+    return ((ordered) dict): the config for the given role. If nothing is defined, it will
+      return an empty dictionary.
+    """
```
Comment (Sabrina85) [inline src/odemis/gui/conf/data.py]: Keep consistent and start always with capital letter after `:`

File: src/odemis/gui/model/__init__.py [hunk 4]
```diff
@@ -119,6 +119,7 @@ class MainGUIData(object):
     # Mapping between the component role and the attribute name on the MainGUIData
     _ROLE_TO_ATTR = {
         "ccd": "ccd",
+        # ccd* -> ccds[]
```
Comment (Sabrina85) [inline src/odemis/gui/model/__init__.py]: what is this line for?

Comment (pieleric) [reply src/odemis/gui/model/__init__.py]: Just to remember that all the "ccd*" roles get mapped to ccds. They is a special code that does it, so it cannot be mapped as the standard roles.

File: src/odemis/gui/conf/data.py (hunk 1 above)
Comment (pieleric) [reply src/odemis/gui/conf/data.py]: We could make it as specific as the regex in the path, but it's an overkill. For now, every role which starts with "photo-detector" will be treated by the GUI as a photo-detector. I think it's fair, and if we want more fine tuning, we'll need to update the code.

**Comment by pieleric** [comment]
Just updated the first commit with some tiny fixes.

**Pull request merged**